frontend/src/app/components/FeaturedExperiment/FeaturedButton.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+import React from 'react';
+import { Localized } from 'fluent-react/compat';
+import { buildSurveyURL, experimentL10nId } from '../../lib/utils';
+
+import Modal from '../Modal';
+import MeasurementSection from '../Measurements';
+import MainInstallButton from '../MainInstallButton';
+
+import type { InstalledExperiments } from '../../reducers/addon';
+
+type FeaturedButtonProps = {
+  clientUUID?: string,
+  enabled: boolean,
+  experiment: Object,
+  eventCategory: string,
+  hasAddon: any,
+  installed: InstalledExperiments,
+  getExperimentLastSeen: Function,
+  sendToGA: Function,
+  postInstallCallback: Function
+}
+
+type FeaturedButtonState = {
+  showLegalDialog: boolean
+}
+
+export default class FeaturedButton extends React.Component {
+  props: FeaturedButtonProps
+  state: FeaturedButtonState
+
+  constructor(props: FeaturedButtonProps) {
+    super(props);
+    this.state = {
+      showLegalDialog: false
+    };
+  }
+
+  l10nId(pieces: string) {
+    return experimentL10nId(this.props.experiment, pieces);
+  }
+
+  renderLegalLink() {
+    const { title } = this.props.experiment;
+
+    const terms = <Localized id="landingLegalNoticeTermsOfUse">
+            <a href="/terms">terms</a>
+          </Localized>;
+    const privacy = <Localized id="landingLegalNoticePrivacyNotice">
+            <a href="/privacy">privacy</a>
+          </Localized>;
+
+    const launchLegalModal = (ev) => {
+      ev.preventDefault();
+      this.setState({ showLegalDialog: true });
+    };
+
+    return (<Localized id={this.l10nId('legal-link')}>
+              <p className="main-install__legal">
+                By proceeding, you agree to the {terms} and {privacy} of Test Pilot
+                and <a href="#" onClick={launchLegalModal}>{title}</a>.
+              </p>
+            </Localized>);
+  }
+
+  handleManage() {
```

Comment:
nit: Might be good to update these to use the class property syntax to get these all bound in the constructor.

So you can have `handleManage = () => { ... }` here and `onClick={this.handleManage}` in the JSX - rather than `onClick={() => this.handleManage()`